src/test/scala/performanceanalysis/administrator/AdministratorSpec.scala

```diff
@@ -58,6 +59,21 @@ class AdministratorSpec extends SpecBase with ScalatestRouteTest {
         result shouldBe Details(componentId)
       }
     }
+
+    "handle a POST on /components by creating a new registered componentId" in new Administrator(system.deadLetters) {
+      val probe = TestProbe("AdministratorActorProbe")
```

Comment:
Since Codacy is complaining about increased code duplication here, it might be a good idea to refactor the probe-with-autopilot functionality towards SpecBase